homeassistant/components/vizio/config_flow.py

```diff
@@ -231,7 +263,98 @@ def __init__(self) -> None:
+
+    async def async_step_pair_tv(
+        self, user_input: Dict[str, Any] = None
+    ) -> Dict[str, Any]:
+        """Start pairing process and ask user for PIN to complete pairing process."""
+        errors = {}
+
+        # Start pairing process if it hasn't already started
+        if not self._ch_type and not self._pairing_token:
+            dev = VizioAsync(
+                DEVICE_ID,
+                self._data[CONF_HOST],
+                self._data[CONF_NAME],
+                None,
+                self._data[CONF_DEVICE_CLASS],
+                session=async_get_clientsession(self.hass, False),
+            )
+            pair_data = await dev.start_pair()
+
+            if pair_data:
+                self._ch_type = pair_data.ch_type
+                self._pairing_token = pair_data.token
+                return await self.async_step_pair_tv()
+
+            return self.async_show_form(
+                step_id="user",
+                data_schema=_get_config_schema(self._data),
+                errors={"base": "cant_connect"},
+            )
+
+        # Complete pairing process if PIN has been provided
+        if user_input and user_input.get(CONF_PIN):
+            dev = VizioAsync(
+                DEVICE_ID,
+                self._data[CONF_HOST],
+                self._data[CONF_NAME],
+                None,
+                self._data[CONF_DEVICE_CLASS],
+                session=async_get_clientsession(self.hass, False),
+            )
+            pair_data = await dev.pair(
+                self._ch_type, self._pairing_token, user_input[CONF_PIN]
+            )
+
+            if pair_data:
+                self._data[CONF_ACCESS_TOKEN] = pair_data.auth_token
+                self._must_show_form = True
+
+                # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+                if self.context["source"] == SOURCE_IMPORT:
+                    # If user is pairing via config import, show different message
+                    return await self.async_step_pairing_complete_import()
+
+                return await self.async_step_pairing_complete()
+
+            # If no data was retrieved, it's assumed that the pairing attempt was not
+            # successful
+            errors[CONF_PIN] = "complete_pairing_failed"
+
+        return self.async_show_form(
+            step_id="pair_tv",
+            data_schema=_get_pairing_schema(user_input),
+            errors=errors,
+        )
+
+    async def _pairing_complete(self, step_id: str) -> Dict[str, Any]:
+        """Handle config flow completion."""
+        if not self._must_show_form:
+            return await self._create_entry_if_unique(self._data)
+
+        self._must_show_form = False
+        return self.async_show_form(
+            step_id=step_id,
+            data_schema=vol.Schema({}),
+            description_placeholders={
+                "access_token_value": self._data[CONF_ACCESS_TOKEN],
```

Comment:
Why would we show the access token to the user ?

Reply:
It was suggested by a user so that the `access_token` could be stored for the future, e.g. in case a user wants to add it to their configuration.yaml to be able to rebuild their instance without having to re-pair

Reply:
@balloob any additional thoughts here?